cis194/week8/tien/src/Party.hs

```diff
@@ -20,3 +22,26 @@ instance Monoid GuestList where
 -- 1.3 --
 moreFun :: GuestList -> GuestList -> GuestList
 moreFun = max
+
+-- exercie 2 --
+treeFold :: (a -> [b] -> b) -> Tree a -> b
+treeFold f (Node l []) = f l []
+treeFold f (Node l s) = f l $ treeFold f <$> s
+
+-- exercie 3 --
+nextLevel :: Employee -> [(GuestList, GuestList)] -> (GuestList, GuestList)
+nextLevel emp gls = (gl1, gl2)
+  where
+    gl1 = glCons emp $ foldr (<>) mempty $ snd <$> gls
```

Comment:
`foldr (<>) mempty` seems like `fold` from `Data.Foldable`. It comes with `base`
```suggestion
    gl1 = glCons emp . fold . fmap snd $ gls
```